rotkehlchen/exchanges/coinbase.py

```diff
@@ -95,6 +95,57 @@ def trade_from_coinbase(raw_trade: Dict[str, Any]) -> Optional[Trade]:
     )
 
 
+def trade_from_conversion(trade_a: Dict[str, Any], trade_b: Dict[str, Any]) -> Optional[Trade]:
+    """Turn information from a conversion into a trade
+
+    Mary raise:
+    - UnknownAsset due to Asset instantiation
+    - DeserializationError due to unexpected format of dict entries
+    - KeyError due to dict entires missing an expected entry
+    """
+    # Check that the status is complete
+    if trade_a['status'] != 'completed':
+        return None
+
+    # Trade b will represent the asset we are converting to
+    if trade_b['amount']['amount'].startswith('-'):
+        trade_a, trade_b = trade_b, trade_a
+
+    timestamp = deserialize_timestamp_from_date(trade_a['updated_at'], 'iso8601', 'coinbase')
+    trade_type = deserialize_trade_type('sell')
+    tx_amount = deserialize_asset_amount(trade_b['amount']['amount'])
+    tx_asset = asset_from_coinbase(trade_b['amount']['currency'], time=timestamp)
+    native_amount = deserialize_asset_amount(trade_b['native_amount']['amount'])
+    native_asset = asset_from_coinbase(trade_b['native_amount']['currency'], time=timestamp)
+    # in coinbase you are buying/selling tx_asset for native_asset
+    pair = TradePair(f'{tx_asset.identifier}_{native_asset.identifier}')
+    amount = tx_amount
+    # The rate is how much you get/give in quotecurrency if you buy/sell 1 unit of base currency
+    rate = Price(native_amount / tx_amount)
+
+    # Obtain fee amount using data from both trades
+    amount_after_fee = trade_b['native_amount']['amount']
+    amount_before_fee = trade_a['amount']['amount']
+    conversion_fee_amount = eval(amount_after_fee + amount_before_fee)
```

Comment:
please do not **ever ever ever** use `eval()`.

Especially as you do here with external input. I missed this in the first review. This is a very very serious security risk.

It's like SQL injection but worse. Someone, or heck even coinbase could, make the remote endpoint return python code in the text of `trade_b['native_amount']['amount']` and the next one. The `eval` will blindly execute it. Any code. Anything. Could be a: "zip_all_user_data();send_to_badguys_server();rm_rf_everything();"

What you should do instead is deserialize each individual field using `deserialize_asset_amount` from `serialization/deserialize.py`

```suggestion
    amount_after_fee = deserialize_asset_amount(trade_b['native_amount']['amount'])
    amount_before_fee = deserialize_asset_amount(trade_a['amount']['amount'])
    conversion_fee_amount = amount_after_fee + amount_before_fee
```

For more on eval read [here](https://stackoverflow.com/questions/1832940/why-is-using-eval-a-bad-practice), or just google why it's a bad idea. There is always a better way to do it.